### PR TITLE
Updating runtimes to include new Node.js v12 image.

### PR DESCRIPTION
**Pull request by jthomas**
## Description

Updated runtimes definition to include new Node.js v12 runtime image.

- Updated runtimes manifest
- Added API documentation
- Minor updates to docs
- Added automated test case

## Related issue and scope

https://github.com/apache/incubator-openwhisk-runtime-nodejs/issues/124

## My changes affect the following components
- [x] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [x] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [x] Documentation

## Types of changes
- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:
- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [x] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: ansible/files/runtimes.json [hunk 1]
```diff
@@ -48,6 +48,20 @@
                     "attachmentName": "codefile",
                     "attachmentType": "text/plain"
                 }
+            },
+            {
+                "kind": "nodejs:12",
+                "default": true,
```
Comment (rabbah) [inline ansible/files/runtimes.json]: should this be the default?

File: docs/actions-nodejs.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ and demonstrate how to bundle multiple JavaScript files and third party dependen
 
   The CLI automatically infers the type of the action by using the source file extension.
   For `.js` source files, the action runs by using a Node.js runtime. You may specify
-  the Node.js runtime to use by explicitly specifying the parameter `--kind nodejs:10`, `--kind nodejs:8`, or `--kind nodejs:6`
+  the Node.js runtime to use by explicitly specifying the parameter `--kind nodejs:12`, `--kind nodejs:10`, `--kind nodejs:8`, or `--kind nodejs:6`
```
Comment (rabbah) [inline docs/actions-nodejs.md]: remove nodejs:6?

File: docs/actions-nodejs.md [hunk 3]
```diff
@@ -222,7 +222,7 @@ zip -r action.zip *
 wsk action create packageAction --kind nodejs:10 action.zip
 ```
 
-When creating an action from a `.zip` archive with the CLI tool, you must explicitly provide a value for the `--kind` flag by using `nodejs:10`, `nodejs:8` or `nodejs:6`.
+When creating an action from a `.zip` archive with the CLI tool, you must explicitly provide a value for the `--kind` flag by using `nodejs:12`, `nodejs:10`, `nodejs:8` or `nodejs:6`.
```
Comment (rabbah) [inline docs/actions-nodejs.md]: same re 6.

File: docs/concurrency.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ Concurrent activation processing within the same action container can be enabled
 
 * enable the akka http client at invoker config
   * e.g. CONFIG_whisk_containerPool_akkaClient=true
-* use a kind that supports concurrency (currently only `nodejs:10`, `nodejs:8` and `nodejs:6`)
+* use a kind that supports concurrency (currently only `nodejs:12`, `nodejs:10`, `nodejs:8` and `nodejs:6`)
```
Comment (rabbah) [inline docs/concurrency.md]: remove 6.

File: docs/actions-nodejs.md (hunk 2 above)
Comment (jthomas) [reply docs/actions-nodejs.md]: I'm not sure about this. It is still available in the project (for back compat reasons atm). Until it's removed - which is another PR, I think we should keep it in the docs?

Comment (csantanapr) [reply docs/actions-nodejs.md]: Remove it from the docs now

**Comment by csantanapr** [comment]
@jthomas 
Conflicting files:
ansible/files/runtimes.json

**Pull request merged**